From df8138c4f6286f1fe3befba3ce4d0e09ab713c80 Mon Sep 17 00:00:00 2001
From: sandstranger <sandstranger@gmail.com>
Date: Sat, 25 Mar 2017 10:01:00 +0400
Subject: [PATCH] From: daniel_hk (https://github.com/daniel_hk) Date Tue, 17
 Jan 2017 15:40:08 +0000 Subject: Skip the empty SSID in scan message

Add the length check to skip empty entries.
---
 src/drivers/driver_nl80211_scan.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/drivers/driver_nl80211_scan.c b/src/drivers/driver_nl80211_scan.c
index c115b6b..25597dc 100644
--- a/src/drivers/driver_nl80211_scan.c
+++ b/src/drivers/driver_nl80211_scan.c
@@ -949,9 +949,10 @@ int wpa_driver_nl80211_vendor_scan(struct i802_bss *bss,
 			wpa_hexdump_ascii(MSG_MSGDUMP, "nl80211: Scan SSID",
 					params->ssids[i].ssid,
 					params->ssids[i].ssid_len);
-			if (nla_put(msg, i + 1, params->ssids[i].ssid_len,
-				    params->ssids[i].ssid))
-				goto fail;
+                                if (params->ssids[i].ssid_len)
+                               if (nla_put(msg, i + 1, params->ssids[i].ssid_len,
+                                           params->ssids[i].ssid))
+                                       goto fail;
 		}
 		nla_nest_end(msg, ssids);
 	}
